// logkitty/build/errors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ERR_CANNOT_START_LOGCAT = exports.ERR_CANNOT_CLEAN_LOGCAT_BUFFER = exports.ERR_CANNOT_GET_APP_PID = exports.ERR_UNPROCESSABLE_PID = exports.CodeError = void 0;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class CodeError extends Error {
  constructor(code, message) {
    super(message);

    _defineProperty(this, "code", void 0);

    this.code = code;
  }

}

exports.CodeError = CodeError;
const ERR_UNPROCESSABLE_PID = 'ERR_UNPROCESSABLE_PID';
exports.ERR_UNPROCESSABLE_PID = ERR_UNPROCESSABLE_PID;
const ERR_CANNOT_GET_APP_PID = 'ERR_CANNOT_GET_APP_PID';
exports.ERR_CANNOT_GET_APP_PID = ERR_CANNOT_GET_APP_PID;
const ERR_CANNOT_CLEAN_LOGCAT_BUFFER = 'ERR_CANNOT_CLEAN_LOGCAT_BUFFER';
exports.ERR_CANNOT_CLEAN_LOGCAT_BUFFER = ERR_CANNOT_CLEAN_LOGCAT_BUFFER;
const ERR_CANNOT_START_LOGCAT = 'ERR_CANNOT_START_LOGCAT';
exports.ERR_CANNOT_START_LOGCAT = ERR_CANNOT_START_LOGCAT;
